build/media_source/system/js/fields/passwordcaps.es6.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @copyright   (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+// Detect Caps Lock status
+function checkCapsLock(event) {
+  const capsLockOn = event.getModifierState && event.getModifierState('CapsLock');
+  const target = event.target;
+  let capsErrorMessage = document.getElementById('capsLockError');
+
+  if (capsLockOn && target.type === 'password') {
+    // Caps Lock is on
+    if (!capsErrorMessage) {
+      // Create error message element if it doesn't exist
+      capsErrorMessage = document.createElement('span');
+      capsErrorMessage.id = 'capsLockError';
+      capsErrorMessage.textContent = Joomla.Text._('JCAPSLOCKON');
+      capsErrorMessage.setAttribute('class', 'invalid form-control-hint');
+      target.parentNode.before(capsErrorMessage);
+    }
+  } else {
+    // Caps Lock is off or not targeting a password input field
+    if (capsErrorMessage) {
+      // Remove error message if it exists
+      capsErrorMessage.parentNode.removeChild(capsErrorMessage);
+    }
+  }
+}
+
+// Attach event listener to password input field
+const passwordInput = document.querySelector('input[type="password"]');
+passwordInput.addEventListener('keydown', checkCapsLock);
+passwordInput.addEventListener('mousedown', checkCapsLock);
```

Comment:
```suggestion
passwordInput.addEventListener('input', checkCapsLock);
```

one event is enough here

Reply:
Could also be `focus`

Reply:
need to double check this as the mousedown was added because otherwise the key is only detected after the first input. But mousedown also added the check when you click

Reply:
FOCUS did not work